backends/rapidpro/msg.go

```diff
@@ -327,24 +327,31 @@ func checkMsgSeen(b *backend, msg *DBMsg) courier.MsgUUID {
 	r := b.redisPool.Get()
 	defer r.Close()
 
-	fingerprint := msg.fingerprint()
+	urnFingerprint := msg.urnFingerprint()
 
 	now := time.Now().In(time.UTC)
 	prev := now.Add(time.Second * -2)
 	windowKey := fmt.Sprintf("seen:msgs:%s:%02d", now.Format("2006-01-02-15:04"), now.Second()/2*2)
 	prevWindowKey := fmt.Sprintf("seen:msgs:%s:%02d", prev.Format("2006-01-02-15:04"), prev.Second()/2*2)
 
-	// try to look up our UUID from either window or prev window
-	foundUUID, _ := redis.String(luaMsgSeen.Do(r, windowKey, prevWindowKey, fingerprint))
-	if foundUUID != "" {
-		return courier.NewMsgUUIDFromString(foundUUID)
+	// see if there were any messages received in the past 4 seconds
+	found, _ := redis.String(luaMsgSeen.Do(r, windowKey, prevWindowKey, urnFingerprint))
+
+	// if so, test whether the text it the same
+	if found != "" {
+		prevText := found[37:]
```

Comment:
I am confused here do we still have the text?

Reply:
we insert the msg uuid & text in the writeMsgSeen

Reply:
I got it